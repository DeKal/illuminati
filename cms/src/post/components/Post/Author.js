import React from 'react'
import PropTypes from 'prop-types'
import TextField from 'post/components/TextField'

const Author = ({ isEdited, author, editPostAuthor }) => (
  <TextField
    label="Author"
    variant="outlined"
    value={author}
    fullWidth
    error={isEdited}
    onChange={event => editPostAuthor(event.target.value)}
  />
)

export default Author

Author.propTypes = {
  isEdited: PropTypes.bool,
  author: PropTypes.string,
  editPostAuthor: PropTypes.func
}
