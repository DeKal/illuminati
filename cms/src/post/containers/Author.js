import { connect } from 'react-redux'
import { getPostAuthor, getEditedAuthor } from 'post/state/selectors'
import { editPostAuthor } from 'post/state/actions'
import Author from 'post/components/Post/Author'

const mapStateToProps = state => {
  const editedAuthor = getEditedAuthor(state)
  const author = getPostAuthor(state)
  return {
    author: editedAuthor || '',
    isEdited: editedAuthor !== author
  }
}

const mapDispatchToProps = {
  editPostAuthor
}

export default connect(mapStateToProps, mapDispatchToProps)(Author)
