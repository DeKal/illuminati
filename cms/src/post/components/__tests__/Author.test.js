import * as React from 'react'
import { shallow } from 'enzyme'
import Author from '../Author'

describe('Author', () => {
  it('should render Author correctly', function() {
    const wrap = shallow(<Author author="author" />)
    expect(wrap).toMatchSnapshot()
  })
})
