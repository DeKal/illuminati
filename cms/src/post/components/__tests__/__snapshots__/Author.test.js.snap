// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author should render Author correctly 1`] = `
<WithStyles(WithStyles(ForwardRef(TextField)))
  fullWidth={true}
  label="Author"
  onChange={[Function]}
  value="author"
  variant="outlined"
/>
`;
